cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -318,38 +320,43 @@ private void restartIfNecessary(int podId, RestartContext restartContext)
         }
 
         try {
-            RestartPlan restartPlan = restartPlan(podId, pod, restartContext);
-            if (restartPlan.needsRestart || restartPlan.needsReconfig) {
-                if (deferController(podId, restartContext)) {
-                    log.debug("{}: Pod {} is controller and there are other pods to roll", reconciliation, podId);
-                    throw new ForceableProblem("Pod " + podName(podId) + " is currently the controller and there are other pods still to roll");
-                } else {
-                    if (canRoll(podId, 60_000, TimeUnit.MILLISECONDS, false)) {
-                        // Check for rollability before trying a dynamic update so that if the dynamic update fails we can go to a full restart
-                        if (!maybeDynamicUpdateBrokerConfig(podId, restartPlan)) {
-                            log.debug("{}: Pod {} can be rolled now", reconciliation, podId);
-                            restartAndAwaitReadiness(pod, operationTimeoutMs, TimeUnit.MILLISECONDS);
+            if (isCrashlooping(pod)) {
```

Comment:
What if the pod is not crash looping but creating or pending because of some other misconfiguration (missing volumes, badly configured resources, ...)? Does the rolling update work fine in that case even with this check looking just for crash-looping pods?

Reply:
How did you addressed this? This still seems to do something only when the pod is crashlooping but not when it gets stuck in other states such as Pending, ContainerCreating, ImagePullBackoff etc.